Lib/typing.py

```diff
@@ -962,7 +959,7 @@ def __mro_entries__(self, bases):
 
     def __getattr__(self, attr):
         if attr in {'__name__', '__qualname__'}:
-            return self._name
+            return self._name or self.__origin__.__name__
```

Comment:
This solution is so nice. I wish I'd thought about this a little deeper when I proposed the original one:(.

Anyways if I'm not wrong, `__origin__` will always be set on all `GenericAlias` to some type, so this should always succeed.